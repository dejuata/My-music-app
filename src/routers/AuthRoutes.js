import React from 'react'
import { Route, Routes } from 'react-router-dom'
import { SignInPage } from '../components/pages/signin/SignInPage'
import { SignUpPage } from '../components/pages/signup/SignUpPage'


export const AuthRoutes = () => {
    return (
        <>
            <Routes>
                {/* <Route path="sign-in" element={<SignInPage />} /> */}
                {/* <Route path="sign-up" element={<SignUpPage />} /> */}

                <Route path="/" element={<SignInPage />} />
            </Routes>
        </>
    )
}