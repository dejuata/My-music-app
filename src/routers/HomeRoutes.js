import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Route, Routes } from 'react-router-dom'
import { setTracks } from '../actions/track'
import { getRecommendedTracks } from '../api/selectors/getRecommendedTracks'
import { Header } from '../components/organisms/Header/Header'
import { Navbar } from '../components/organisms/Navbar/Navbar'
import { FavoritesPage } from '../components/pages/favorites/FavoritesPage'
import { HomePage } from '../components/pages/home/HomePage'
import { TrackPage } from '../components/pages/track/TrackPage'
import ReactLoading from "react-loading";

export const HomeRoutes = () => {

    const [loading, setLoading] = useState(true)
    const dispatch = useDispatch();

    useEffect(() => {

        getRecommendedTracks('latin')
            .then( data => {
                dispatch( setTracks(data) );
                setLoading(false);
            })

    }, [ dispatch ])

    return (
        <>
            <Header />
            <div className='home__container'>
            {

                loading
                ?
                    <div className='home__loading'>
                        <ReactLoading type={ 'bubbles' } color="#fff" />
                    </div>
                :
                    <Routes>
                        <Route path="/" element={<HomePage />} />
                        <Route path="favorites" element={<FavoritesPage />} />
                        <Route path="track/:trackId" element={<TrackPage />} />

                    </Routes>

            }
            </div>

            <div className='home__navbar'>
                <Navbar />
            </div>
        </>
    )
}