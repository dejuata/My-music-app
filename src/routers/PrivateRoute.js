import React from 'react';
import PropTypes from 'prop-types';

import { Navigate } from 'react-router-dom';


export const PrivateRoute = ({
    isAuthenticated,
    children: Component,
}) => {

    return ( isAuthenticated )
        ? ( Component )
        : ( <Navigate to="/auth/login" /> )
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    children: PropTypes.object.isRequired
}
