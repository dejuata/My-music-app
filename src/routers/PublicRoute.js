import React from 'react';
import PropTypes from 'prop-types';

import { Navigate } from 'react-router-dom';


export const PublicRoute = ({
    isAuthenticated,
    children: Component,
}) => {

    return ( isAuthenticated )
        ? ( <Navigate to="/" /> )
        : ( Component )
}

PublicRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    children: PropTypes.object.isRequired
}
