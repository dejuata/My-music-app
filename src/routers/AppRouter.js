import { getAuth, onAuthStateChanged } from 'firebase/auth';
import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { Route, Routes, BrowserRouter, Navigate } from 'react-router-dom';
import { login } from '../actions/auth';
import { AuthRoutes } from './AuthRoutes';
import { HomeRoutes } from './HomeRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const [checking, setChecking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false)
    const { logged } = useSelector(state => state.auth);

    useEffect(() => {
        if (logged) {
            setIsLoggedIn(true);
        } else {
            setIsLoggedIn(false);
        }
        setChecking(false)
    }, [logged])

    if (checking) {
        return (
            <h2>Loading...</h2>
        )
    }

    return (
        <BrowserRouter>
            <Routes>

                <Route path="/auth/*" element={
                    <PublicRoute isAuthenticated = { isLoggedIn }>
                        <AuthRoutes />
                    </PublicRoute>
                } />

                <Route path="/*" element={
                    <PrivateRoute isAuthenticated = { isLoggedIn }>
                        <HomeRoutes />
                    </PrivateRoute>
                } />

                <Route path="/*" element={<Navigate to="/auth/login" />} />

            </Routes>
        </BrowserRouter>
    )
}
