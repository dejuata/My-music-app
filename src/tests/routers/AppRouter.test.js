import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import { AppRouter } from '../../routers/AppRouter';


const middlewares = [ thunk ]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);

const wrapper = mount(
    <Provider store={ store }>
            <AppRouter />
    </Provider>
)

describe('Test AppRouter', () => {

    test('Should be render AppRouter', () => {

        expect(wrapper).toMatchSnapshot();

    })


})