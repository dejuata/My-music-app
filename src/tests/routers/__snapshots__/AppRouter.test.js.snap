// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppRouter Should be render AppRouter 1`] = `
<div
  className="auth__container"
>
  <div
    className="auth-form__container"
  >
    <div
      className="a-logo-container"
    >
      <div
        className="a-logo-container__image"
      >
        <img
          alt="logo"
          src="/assets/logo.svg"
        />
      </div>
      <p
        className="a-logo-container__title"
      >
        MY MUSIC APP
      </p>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-control mb-2"
      >
        <label
          className="label"
        >
          Email address
        </label>
        <input
          autoComplete="off"
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="example@email.com"
          type="email"
          value="demo@gmail.com"
        />
      </div>
      <div
        className="form-control mb-3"
      >
        <label
          className="label"
        >
          Password
        </label>
        <input
          autoComplete="off"
          className="input"
          name="password"
          onChange={[Function]}
          placeholder="********"
          type="password"
          value="123456"
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Log in
      </button>
      <a
        className="auth__link"
        href="/auth/register"
        onClick={[Function]}
      >
        Create new account
      </a>
    </form>
    <div
      className="auth__button"
    >
      <a
        href="https://accounts.spotify.com/authorize?client_id=13acf932b40845cc882aab4525a8df81&response_type=token&redirect_uri=http://localhost:3000/auth/redirect&scope=user-library-modify%20user-library-read&show_dialog=true"
      >
        <div
          className="btn btn-secondary"
        >
          Log In with Spotify
        </div>
      </a>
    </div>
  </div>
</div>
`;
